controllers/comments.controller.js

```diff
@@ -5,23 +5,58 @@ pertaining to manipulating comments
 
 */
 
-const { fetchComments } = require("../models/comments.model");
+const { fetchComments, insertComment } = require("../models/comments.model");
 const { fetchArticleById } = require("../models/articles.model");
+const { fetchUserByName } = require("../models/general.model");
 
 const getComments = (request, response, next) => {
   const { article_id } = request.params;
   if (article_id) {
 
     fetchArticleById(article_id)
     .then((article) =>{
-            fetchComments(article_id).then((comments) => {
-              response.status(200).send({ comments });
-            });
+            fetchComments(article_id)
+              .then((comments) => {
+                response.status(200).send({ comments });
+              })
+              .catch((err) => {
+                console.log(err);
+                next(err);
+              });
     })
       .catch((err) => {
         next(err);
       });
   }
 };
 
-module.exports = { getComments };
+const postComments = (request, response, next) =>{
+  const { article_id } = request.params;
+  const {body} = request
+  if (article_id) {
+    fetchArticleById(article_id)
+    .then((article) =>{
+        fetchUserByName(body.username)
+          .then((user) => {
+            insertComment(article_id, body)
+              .then((comment) => {
+                response.status(201).send({ comment });
+              })
+              .catch((err) => {
+                console.log(err);
+                next(err);
+              });
+          })
+          .catch((err) => {
+            console.log(err);
+            next(err);
+          });
+    })
+    .catch((err) => {
+        console.log(err);
+      next(err);
+    });
+}
```

Comment:
you should only need one catch block at the end :) 

Also, you're currently using multiple .then blocks inside each other - realistically, they should chain onto the end of the next one. My advice would be that author in the comments table is a foreign key that relates to username in the users table, so a check isn't needed (a psql error will be thrown!). Plus, article_id is also a foreign key, so again a psql error will be thrown and a check is not needed :)